app/models/dish.rb

```diff
@@ -16,8 +16,41 @@ class Dish < ActiveRecord::Base
   RATE_DIMENSION_DISH = 'dish'
   ratyrate_rateable RATE_DIMENSION_DISH
 
+  def rates
+    Rate.where(rateable_id: id, dimension: RATE_DIMENSION_DISH)
+  end
+
   def raters_count
-    Rate.where(rateable_id: id, dimension: RATE_DIMENSION_DISH).pluck(:rater_id).uniq.count
+    rates.pluck(:rater_id).uniq.count
+  end
+
+  def rating
+    rates.average(:stars) || -1
+  end
+
+  def search_data
+    if user && user.location && user.location.is_visible
```

Comment:
This condition looks similar to a condition in `ApplicationHelper#dish_distance`. I think there's a method (or more) ready to be discovered to encapsulate this.

Perhaps a `user.visible_location` method is one possibility

```ruby
class User ...
  ...
  def visible_location
    location.try(:is_visible) ? location : nil
  end
  ...
end
```